src/framebuffer/swtfb_client.rs

```diff
@@ -0,0 +1,227 @@
+//! This implements the IPC part of a rM2Framebuffer client to interact with this server:
+//! https://github.com/ddvk/remarkable2-framebuffer
+//!
+//! The client is developed according to the spec here:
+//! https://github.com/ddvk/remarkable2-framebuffer/issues/11
+
+use super::mxcfb::mxcfb_update_data;
+use crate::framebuffer::screeninfo::{FixScreeninfo, VarScreeninfo};
+use log::warn;
+use memmap2::{MmapOptions, MmapRaw};
+use std::ffi::{c_void, CString};
+use std::fs::{File, OpenOptions};
+use std::io::Error as IoError;
+use std::os::unix::prelude::AsRawFd;
+use std::{env, mem, ptr};
+
+const SWTFB_MESSAGE_QUEUE_ID: i32 = 0x2257c;
+
+pub const WIDTH: i32 = 1404;
+pub const HEIGHT: i32 = 1872;
```

Comment:
Why not reuse `common::DISPLAYHEIGHT`?

Reply:
Since that could become conditional to a model if a potential Gen 3 changes the resolution (as already happened to MT and Wacom sizes). While this code is duplicated, this should prevent potential problems should such a change occur. And if they continue using a sw based fb, the client would need an update/newer client anyway.

From a technical point, DISPLAYWIDTH should already be a conditional value (as I [learned from reStream](https://github.com/rien/reStream/blob/9deee838/reStream.sh#L131)). libremarkable just always "hid" those last 4 pixels and got away with it.

In that case this would also break, since a const can't be derived from a static once_cell. We actually could address this in a future release.

Reply:
I'd rather we don't repeat magic numbers when we have `crate::framebuffer::common::DISPLAYHEIGHT` today.

Reply:
Good point. Should be done with commit a4e3fd58.